File: booklibrary-app/src/app/models/book.ts
import { BookLoan } from "./book-loan";

export class Book {
  public number?: number;
  public title!: string;
  public author!: string;
  public currentLoan?: BookLoan;
  public loansCount?: number;

  public isLent(): boolean {
    return this.currentLoan ? true : false;
  }
}

File: BookLibrary.Domain/Book.cs
namespace BookLibrary.Domain
{
    public class Book
    {
        public int Number { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public BookLoan CurrentLoan { get; set; }
        public int LoansCount { get; set; }
    }
}